### ISSUE TITLE
Why softmax before argmax?

### ISSUE THREAD

**Issue by vedantroy** [milesial/Pytorch-UNet#379]
Hi! Quick question, isn't the softmax here:
https://github.com/milesial/Pytorch-UNet/blob/e36c782fbfc976b7326182a47dd7213bd3360a7e/train.py#L133

not needed because we do an argmax right after and softmax never changes the argmax?

**Comment by milesial** [comment]
Yes you're right, it doesn't do anything. If you want performance, you can remove it. But also if you want performance, you would not use image logging in wandb.
